lib/driver.ts

```diff
@@ -479,18 +478,29 @@ export class EspressoDriver extends AndroidDriver implements ExternalDriver<
     await this.addDeviceInfoToCaps();
   }
 
+  /**
+   * Turn on or off animation scale.
+   * '--no-window-animation' instrument argument for Espresso disables window animations,
+   * but it does not bring the animation scale back to the pre-instrument process start state in Espresso
+   * unlike Appium UIA2 driver case. We want to disable/enable the animation scale only in an appium espresso session as possible.
+   * @param isEnabled
+   */
   async setWindowAnimationState(isEnabled: boolean): Promise<void> {
     const isAnimationOn = await this.adb.isAnimationOn();
     const shouldDisableAnimation = !isEnabled && isAnimationOn;
     const shouldEnableAnimation = isEnabled && !isAnimationOn;
 
     if (shouldDisableAnimation) {
       this.log.debug('Disabling window animation as "disableWindowAnimation" capability is set to true/fallback to default value "true"');
-      await this.settingsApp.setAnimationState(false);
+      await this.adb.getApiLevel() < 26
+        ? await this.settingsApp.setAnimationState(false)
```

Comment:
would it make sense to move this logic to settingsApp module?

Reply:
Well, i initially thought the same thing but:
- this is only for espresso driver. uia2 driver does it via `--no-window-animation`
     - So uia2 driver won't run `await this.adb.setAnimationScale` to manage the animation stuff only via `--no-window-animation`
- the settingsApp's animation thing should be removed after dropping api level 26, then this disabling/enabling animation thing won't need to be part of the settingsApp
    - Then, this `await this.adb.setAnimationScale` should be only in the espresso driver

So, this is kind of espresso driver specific behavior, thus I haven't added it in the settings module

Reply:
I don't think we need to drop the stuff while we support older android APIs. Same has been already done to wifi-related methods: https://github.com/appium/io.appium.settings/blob/a806a2e1e46a45eeb07a0ccf0a9d9e305e3ba734/lib/commands/network.js#L25

Reply:
Okay -> https://github.com/appium/io.appium.settings/pull/176